### PR TITLE
SALTO-1792: Changed supportedTypes to a map between the wrapper type to the real type and made it mandatory

### PR DESCRIPTION
**Pull request by alonstern**
This is the first PR for changing the structure of the fetch config

- Changed supportedTypes to a map between the wrapper type to the real type
- Made supportedTypes mandatory
- Added support for supportedTypes in adapter that uses `dependsOn`
- Added for each adapter its mapping

---
_Release Notes_: 
None

---
_User Notifications_: 
_Jira Adapter_:
- Some unused types will be removed from the workspace

_Zuora Adapter_:
- Some unused types will be removed from the workspace

### CODE REVIEW

File: packages/zendesk-support-adapter/src/config.ts [hunk 1]
```diff
@@ -1535,6 +1535,45 @@ export const DEFAULT_INCLUDE_ENDPOINTS: string[] = [
   'workspaces',
 ]
 
+export const SUPPORTED_TYPES = {
```
Comment (alonstern) [inline packages/zendesk-support-adapter/src/config.ts:1538]: @noamhammer could you verify me with this mapping?

**Review by netama — COMMENTED** [review]
LGTM with a few nits and a question

File: packages/adapter-components/src/elements/swagger/type_elements/element_generator.ts [hunk 2]
```diff
@@ -327,26 +328,30 @@ export const generateTypes = async (
...
+    .map(def => def.type)
 
-    return {
-      allTypes: _.keyBy(filteredTypes, type => type.elemID.name),
-      parsedConfigs,
-    }
-  }
+  const typesToFetch = [
+    ...Object.values(supportedTypes).flat(),
+    ...Object.keys(supportedTypes).flat(),
```
[the remark this answers is not in the document]
Comment (alonstern) [reply packages/adapter-components/src/elements/swagger/type_elements/element_generator.ts:338]: In Zuora, adding the values (.e.g, `AccountingRules`) does not add the keys (e..g, `settings_AccountingRules`)

**Comment by alonstern** [comment]
Weirdly this is backward compatible with the older supportedTypes values (relevant for stripe)

File: packages/zendesk-support-adapter/src/config.ts [hunk 3]
```diff
@@ -1535,6 +1535,45 @@ export const DEFAULT_INCLUDE_ENDPOINTS: string[] = [
...
+  routing_attribute: ['routing_attributes'],
+  sharing_agreement: ['sharing_agreements'],
+  sla_policy: ['sla_policies'],
+  sla_policy_definition: ['sla_policies_definitions'],
+  support_address: ['support_addresses'],
+  target: ['targets'],
+  ticket_field: ['ticket_fields'],
+  ticket_form: ['ticket_forms'],
+  trigger_category: ['trigger_categories'],
+  trigger_definitions: ['trigger_definitions'],
```
Comment (noamhammer) [inline packages/zendesk-support-adapter/src/config.ts:1569]: I think it should be `trigger_definition: ['trigger_definitions'],`, no?

**Review by netama — APPROVED** [review]
LGTM, but I think the existing stripe config will break in the next PR (right now we're only looking at the values and since an array is the same as an object with indexes as keys it's all working), so I think we will need a solution for it before adding the new functionality

**Comment by netama** [comment]
oh and let's add these to the system_config docs?

**Pull request merged**